dspace-server-webapp/src/main/java/org/dspace/app/rest/ScriptRestController.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.ScriptRest;
+import org.dspace.app.rest.model.hateoas.ProcessResource;
+import org.dspace.app.rest.repository.ScriptRestRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller takes care of all the requests to the system/scripts endpoint
+ */
+@RestController
+@RequestMapping("/api/" + ScriptRest.CATEGORY + "/" + ScriptRest.PLURAL_NAME)
+public class ScriptRestController {
```

Comment:
Again, I'm not sure this Controller needs to exist.  I think we can use `ScriptRest` and `ScriptRestRepository` instead...but maybe I'm missing something here?  Generally speaking, we should ONLY create Controllers for special needs/use cases, otherwise, follow the same development pattern we've been using for all other endpoints.

Reply:
Please refer to the previous comment: https://github.com/DSpace/DSpace/pull/2529#discussion_r336980015